src/Components/WebAssembly/WebAssembly/src/Hosting/RootComponentTypeCache.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Hosting
+{
+    internal static class RootComponentTypeCache
+    {
+        private static readonly Dictionary<Key, Type> _typeToKeyLookUp = new ();
+
+        public static Type GetRootComponent(string assembly, string type)
+        {
+            var key = new Key(assembly, type);
+            if (_typeToKeyLookUp.TryGetValue(key, out var resolvedType))
+            {
+                return resolvedType;
+            }
+            else
```

Comment:
nit, you don't need an else

Reply:
Removed, switched it to a unified shared cache type